src/System.Reflection/tests/FieldInfo/FieldInfo_MethodTests.cs

```diff
@@ -86,197 +39,123 @@ public void TestGetValue_Exception1()
         public void TestGetValue_Exception3()
         {
-            String fieldname = "nonstatic_strField";
-            FieldInfo fi = GetField(fieldname);
-            Object objInstance = new Object();
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-            Assert.Throws<System.ArgumentException>(() => { string retValue = (String)fi.GetValue(objInstance); });
-        }
-
-        //Verify SetValue method for static int FieldType
-        [Fact]
-        public void TestSetValue_StaticIntField()
-        {
-            String fieldname = "intFieldStatic";
-            int origfieldvalue = 0;
-            int newfieldValue = 1000;
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (int)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(newfieldValue));
-
-            //reset static field value to its original value
-            FieldInfoMethodTests.intFieldStatic = origfieldvalue;
-        }
-
-        //Verify SetValue method for non-static int FieldType
-        [Fact]
-        public void TestSetValue_IntField()
-        {
-            String fieldname = "intFieldNonStatic";
-            int origfieldvalue = 0;
-            int newfieldValue = 1000;
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (int)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(newfieldValue));
-        }
-
-        //Verify SetValue method for static String FieldType
-        [Fact]
-        public void TestSetValue_StaticStrField()
-        {
-            String fieldname = "static_strField";
-            String origfieldvalue;
-            String newfieldValue = "new";
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (String)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(newfieldValue));
-
-            //reset static field value to its original value
-            FieldInfoMethodTests.static_strField = origfieldvalue;
+            FieldInfo fi = GetAndTestFieldInfo("nonstatic_strField");
+            object objInstance = new object();
+            Assert.Throws<ArgumentException>(() => { string retValue = (string)fi.GetValue(objInstance); });
```

Comment:
Nit: also, the ```string retValue = ``` isn't actually needed here... this can just be:
```C#
Assert.Throws<ArgumentException>(() => (string)fi.GetValue(objInstance));
```